### ISSUE TITLE
Lint/MixedRegexpCaptureTypes: Do not mix named captures and numbered captures in a Regexp literal

### ISSUE THREAD

**Issue by JakubSQ** [rubocop/rubocop#10816]
## Expected behavior

Rubocop should run without errors

## Actual behavior

Rubocop raise an error

## Steps to reproduce the problem

Here is my regex expression
`content = content.gsub(%r{(?<=<div>)(<br>)+|(<br>)+(?=</div>)})`

In return I receive
`Lint/MixedRegexpCaptureTypes: Do not mix named captures and numbered captures in a Regexp literal.`

IMHO regex treats (? ...) as a group but it's only look behind expression but I am not sure about it.
Is there a way to solve this issue?

## RuboCop version

```1.29.1 (using Parser 3.1.2.0, rubocop-ast 1.18.0, running on ruby 3.1.2 arm64-darwin21)```

**Comment by dvandersluis** [comment]
Looks like there's a bug in `regexp_parser` that is detecting a backreference as a named capture group. The master version of that gem seems to have fixed this bug but it is not released yet.
